torch/csrc/distributed/c10d/ProcessGroup.cpp

```diff
@@ -73,11 +73,11 @@ ProcessGroup::Work::Work(
       if (inputTensors) {
         inputs.reserve(inputTensors->size());
         for (const auto& tensor : *inputTensors) {
-          inputs.push_back(tensor);
+          inputs.emplace_back(tensor);
         }
       }
       recordingFunction->before(profilingTitle, inputs);
-      std::function<void()> end_handler = [this, recordingFunction]() {
```

Comment:
@pritamdamania87 do you remember why `this` was captured in a lambda? (Looks like it is unused)